DQM/DTMonitorModule/src/DTCalibValidationFromMuons.cc

```diff
@@ -0,0 +1,374 @@
+ *
+ *  \author G. Mila - INFN Torino
+ */
+
+#include "DQM/DTMonitorModule/interface/DTCalibValidationFromMuons.h"
+
+// Framework
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/EventSetup.h"
+#include "FWCore/Framework/interface/ESHandle.h"
+#include "FWCore/ServiceRegistry/interface/Service.h"
+#include "DQMServices/Core/interface/DQMStore.h"
+#include "DQMServices/Core/interface/MonitorElement.h"
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+
+//Geometry
+#include "Geometry/DTGeometry/interface/DTGeometry.h"
+#include "Geometry/Records/interface/MuonGeometryRecord.h"
+
+//RecHit
+#include "DataFormats/DTRecHit/interface/DTRecSegment4DCollection.h"
+#include "DataFormats/DTRecHit/interface/DTRecHitCollection.h"
+#include "DataFormats/MuonReco/interface/MuonSelectors.h"
+
+
+#include <iterator>
+
+using namespace edm;
+using namespace std;
+
+
+DTCalibValidationFromMuons::DTCalibValidationFromMuons(const ParameterSet& pset) {
+
+  parameters = pset;
+
+  // the name of the 4D segments
+  segment4DToken_ = consumes<DTRecSegment4DCollection>(
+      edm::InputTag(parameters.getUntrackedParameter<string>("segment4DLabel")));
+  // muon collection for matching 4D segments to muons
+  muonToken_ = consumes<reco::MuonCollection>(
+      edm::InputTag(parameters.getUntrackedParameter<string>("muonLabel")));
+  // the counter of segments not used to compute residuals
+  wrongSegment = 0;
+  // the counter of segments used to compute residuals
+  rightSegment = 0;
+
+  nevent=0;
+
+}
+
+
+DTCalibValidationFromMuons::~DTCalibValidationFromMuons(){
+
+  //FR the following was previously in the endJob
+
+ LogVerbatim("DTCalibValidationFromMuons") << "Segments used to compute residuals: " << rightSegment;
+ LogVerbatim("DTCalibValidationFromMuons") << "Segments not used to compute residuals: " << wrongSegment;
+
+}
+
+void DTCalibValidationFromMuons::dqmBeginRun(const edm::Run& run, const edm::EventSetup& setup) {
+
+ // get the geometry
+  setup.get<MuonGeometryRecord>().get(dtGeom);
+}
+
+
+void DTCalibValidationFromMuons::analyze(const edm::Event& event, const edm::EventSetup& setup) {
+
+  ++nevent;
+  LogTrace("DTCalibValidationFromMuons") << "[DTCalibValidationFromMuons] Analyze #Run: " << event.id().run()
+                                << " #Event: " << nevent;
+
+  // RecHit mapping at Step 3 ---------------------------------
+  LogTrace("DTCalibValidationFromMuons") << "  -- DTRecHit S3: begin analysis:";
+  // Get the 4D rechits from the event
+  Handle<reco::MuonCollection> muonH;
+  event.getByToken(muonToken_, muonH);
+  const vector<reco::Muon>* muons = muonH.product();
+
+  // Get the 4D rechits from the event
+  Handle<DTRecSegment4DCollection> segment4Ds;
+  event.getByToken(segment4DToken_, segment4Ds);
+
+  vector<const DTRecSegment4D*> selectedSegment4Ds;
+
+  for (auto &imuon : *muons) {
+    for (const auto &ch : imuon.matches()) {
+        DetId chId(ch.id.rawId());
```

Comment:
The indentation in this file seems to be a bit inconsistent. Could you please review it manually or run it through clang format?

Reply:
We have updated the file with clang format